packages/wrangler/src/pages/validate.ts

```diff
@@ -124,9 +127,9 @@ export const validate = async (args: {
 
 	const fileMap = await walk(directory);
 
-	if (fileMap.size > MAX_ASSET_COUNT) {
+	if (fileMap.size > fileCountLimit) {
 		throw new FatalError(
-			`Error: Pages only supports up to ${MAX_ASSET_COUNT.toLocaleString()} files in a deployment. Ensure you have specified your build output directory correctly.`,
+			`Error: Pages only supports up to ${fileCountLimit.toLocaleString()} files in a deployment. Ensure you have specified your build output directory correctly.`,
```

Comment:
Since this now varies depending on plan, maybe we should say `Pages only supports up to N files in a deployment for your current account plan` or similar. Also, are we sure we don't need `cloudflare-docs` change as well?